### PR TITLE
Display error message from setup submit

### LINKED ISSUE #10926: Setup Page does not show any error messages
I got my database into a weird state by running tests so when I started the web server again it put me back into the setup flow.

After getting to the last step, I clicked the `Next` button, and... nothing happened. The API call to `POST /api/setup` failed with a 500 because I was trying to insert a User with a duplicate email. The API endpoint returns an error message saying as much; the API response looks something like

```js
{"message": "User with email address cam@example.com already exists."}
```

The backend logs helpfully logged this message but it is ignored entirely on the frontend. 

Under normal circumstances the API endpoint shouldn't fail, but if it does, we absolutely need to show the error message returned by the endpoint.

Marking this as P1 because [it's "anything related to Setup"](https://github.com/metabase/metabase/wiki/Bug-Prioritization)

### PR DESCRIPTION
**Pull request by paulrosenzweig**
Resolves #10926

This adds an error message for the big submission at the end of the setup process. My assumption is that any easily handled errors were presented to the user earlier (e.g. bad password, db config issue).

This tries grab a message out of any errored response and plops it next to the "Next" button.
Specifically, it pulls out a `message` or the first value in an `errors` object. 

In my forced errors (below) the messages were either confusing (token isn't a user-facing concept) or really ugly (a raw SQL error response), but I figure it's better to give the user _something_ google-able rather than a generic error or silence.

I started writing a test for this, but for a one-off component like this I figure it doesn't really add much.

### CODE REVIEW

**Comment by mazameli** [comment]
Can we improve these error messages as part of this, or is there an indeterminate list of error messages? I would have no idea what to do with these (other than google them like you said).

**Comment by paulrosenzweig** [comment]
@mazameli I'm thinking of this more as a last ditch effort to give the users something when we couldn't display a more useful error. 

If we want to we can update error messages but the real list is indeterminate. The really ugly one from the db was returned with a 500 from the server.

The [token one](https://github.com/metabase/metabase/blob/fix-10926/src/metabase/api/setup.clj#L36) was intentionally returned, but I'm not sure we should bother updating that. I'm not sure how that would happen unless it was forced in dev like I did.

Hopefully, if any errors are regular enough, people will open issues and we can present more useful info.

File: frontend/src/metabase/setup/components/PreferencesStep.jsx [hunk 1]
```diff
@@ -31,7 +33,12 @@ export default class PreferencesStep extends Component {
     e.preventDefault();
 
     // okay, this is the big one.  we actually submit everything to the api now and complete the process.
-    this.props.submitSetup();
+    const { payload } = await this.props.submitSetup();
+    console.log("payload", payload);
```
Comment (kdoh) [inline frontend/src/metabase/setup/components/PreferencesStep.jsx]:
```suggestion

```

**Review by kdoh — COMMENTED** [review]
One stray `console.log` hanging out in there but otherwise looks good.

**Comment by kdoh** [comment]
Might also want to switch the base branch.

**Review by kdoh — APPROVED** [review]

**Pull request merged**